connectorx/src/sources/postgres/mod.rs

```diff
@@ -961,3 +965,745 @@ impl_produce!(
+            impl<'r> Produce<'r, $t> for PostgresSimpleSourceParser {
+                type Error = PostgresSourceError;
+
+                #[throws(PostgresSourceError)]
+                fn produce(&'r mut self) -> $t {
+                    let (ridx, cidx) = self.next_loc()?;
+                    let val = match &self.rows[ridx] {
+                        SimpleQueryMessage::Row(row) => match row.try_get(cidx)? {
+                            Some(s) => s
+                                .parse()
+                                .map_err(|_| ConnectorXError::cannot_produce::<$t>(Some(s.into())))?,
+                            None => throw!(anyhow!(
+                                "Cannot parse NULL in NOT NULL column."
+                            )),
+                        },
+                        SimpleQueryMessage::CommandComplete(c) => {
+                            panic!("get command: {}", c);
+                        }
+                        _ => {
+                            panic!("what?");
+                        }
+                    };
+                    val
+                }
+            }
+
+            impl<'r, 'a> Produce<'r, Option<$t>> for PostgresSimpleSourceParser {
+                type Error = PostgresSourceError;
+
+                #[throws(PostgresSourceError)]
+                fn produce(&'r mut self) -> Option<$t> {
+                    let (ridx, cidx) = self.next_loc()?;
+                    let val = match &self.rows[ridx] {
+                        SimpleQueryMessage::Row(row) => match row.try_get(cidx)? {
+                            Some(s) => Some(
+                                s.parse()
+                                    .map_err(|_| ConnectorXError::cannot_produce::<$t>(Some(s.into())))?,
+                            ),
+                            None => None,
+                        },
+                        SimpleQueryMessage::CommandComplete(c) => {
+                            panic!("get command: {}", c);
+                        }
+                        _ => {
+                            panic!("what?");
+                        }
+                    };
+                    val
+                }
+            }
+        )+
+    };
+}
+
+impl_simple_produce!(i8, i16, i32, i64, f32, f64, Decimal, Uuid, bool,);
+impl_simple_produce_unimplemented!(
+    Value,
+    HashMap<String, Option<String>>,);
+
+impl<'r> Produce<'r, &'r str> for PostgresSimpleSourceParser {
+    type Error = PostgresSourceError;
+
+    #[throws(PostgresSourceError)]
+    fn produce(&'r mut self) -> &'r str {
+        let (ridx, cidx) = self.next_loc()?;
+        let val = match &self.rows[ridx] {
+            SimpleQueryMessage::Row(row) => match row.try_get(cidx)? {
+                Some(s) => s,
+                None => throw!(anyhow!("Cannot parse NULL in non-NULL column.")),
+            },
+            SimpleQueryMessage::CommandComplete(c) => {
+                panic!("get command: {}", c);
```

Comment:
Also remember to throw error instead of panic here and below (with a better error message for example "got unknown simple query message ...").